README.md

```diff
@@ -207,23 +220,273 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+            class: Entity
+                function: get
+                function: privateGet
+            class: File
+                function: get
+                function: privateGet
+            class: Folder
+                function: get
+                function: privateGet
+```
+
+Now let's try a more complicated example where we want to expose the following functions and classes from `synapseclient.entity` module:
+```
+  class: File
+      function: get
+  class: Folder
+      function: get
+```
+
+Note that we do not want the following from the `synapseclient.entity` module:
+* `function: privateGet` from any of the classes
+* `class: Entity`
+
+First, define your `selectClasses` in the shared .R file:
+```r
+methodsToOmit <- "privateGet"
+classToSkip <- "Entity"
+selectClasses <- function(class) {
+    if (any(class$name == classToSkip)) {
+        return(NULL)
+    }
+    if (!is.null(class$methods)) {
+        culledMethods <- lapply(X = class$methods, function(x) {
+            if (any(x$name == methodsToOmit)) NULL else x;
+        })
+        # Now remove the nulls
+        nullIndices <- sapply(culledMethods, is.null)
+        if (any(nullIndices)) {
+            class$methods <- culledMethods[-which(nullIndices)]
+        }
+    }
+    class
+}
+```
+
+Then call `generateRdFiles` and `generateRWrappers` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseclient",
+                module = "synapseclient.entity",
+                modifyClasses = selectClasses)
+```
+```r
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.entity",
+                  setGenericCallback = callback,
+                  modifyClasses = selectClasses)
+```
+
+### Expose functions within a singleton object
+
+In rare cases, we want to expose Python functions under `synapseclient.client.Synapse` without exposing the `Synapse` object itself.
+
+In a Python session, we would do the following:
+```
+  syn = synapseclient.Synapse()
+  syn.login()
+  syn.get()
+```
+
+And in R, users would not need to know about the `Synapse` object:
+```
+  synLogin()
+  synGet()
+```
+
+Note that:
+* all function calls in R access the same underlying Python object `Synapse`
+* the functions name in R has a `syn` prefix
+
+To do this, one would specifying `pyObjectName` and `functionPrefix` params as following:
+
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseclient",
+                module = "synapseclient.client.Synapse",
+                pyObjectName = "syn",
+                functionPrefix = "syn")
+```
+```r
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.client.Synapse",
+                  setGenericCallback = callback,
+                  pyObjectName = "syn",
+                  functionPrefix = "syn")
+```
+
+Then in the same `.onLoad`, create the Python object:
+```r
+pyImport("synapseclient")
+pyExec("syn = synapseclient.Synapse()")
```

Comment:
need to add that the 'syn' in 'syn = synapseclient.synaspe()' has to match the value passed in as pyObjectName.  
(Should we change pyObjectName to pySingletonName?)